### DISCUSSION TITLE
How to invalidate/refresh a page?

### DISCUSSION THREAD

**Discussion by Symbai** [dotnet/maui#8545] [Q&A] [answered]
Is there a method I can call to force refreshing/redrawing the page? I've noticed a couple of bugs/issues with controls being not visible when they should (they take their available space but they are just hidden). I've noticed that these bugs are gone as soon as the user interacts with the page, for example when he presses a button which opens a prompt, he clicks on cancel then those controls magically appears.

On WPF each control including pages have a method call Invalidate() which forces the page to be redrawn. Is there something similar on MAUI?

**Comment by xtuzy** [comment, marked as answer]
invalidatemeasure can update measure, if you want custom draw, you can use graphicsview, it have invalidate

**Reply by Symbai** [reply]
invalidatemeasure does the trick, thank you

**Reply by headintheclouds21** [reply]
Works like a charm. Much appreciated.
